src/components/pill/pill.component.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import StyledPill from './pill.style';
+import Icon from '../icon/icon';
+import { validProps } from '../../utils/ether/ether';
+import tagComponent from '../../utils/helpers/tags/tags';
+
+class Pill extends React.Component {
+  static safeProps = ['onClick']
+
+  renderCloseIcon() {
+    const { onDelete } = this.props;
+    return (
+      <button
+        type='button'
+        onClick={ onDelete }
+        data-element='close'
+      >
+        <Icon type='cross' bgSize='small' />
+      </button>
+    );
+  }
+
+  render() {
+    const {
+      fill,
+      as,
+      onDelete,
+      colourVariant,
+      children
+    } = this.props;
+    return (
+      <StyledPill
+        { ...validProps(this) }
+        inFill={ fill }
+        styledAs={ colourVariant || as }
+        isDeletable={ onDelete }
+        { ...tagComponent('pill', this.props) }
+      >
+        { children }
+        { onDelete && this.renderCloseIcon() }
+      </StyledPill>
+    );
+  }
+}
+
+Pill.propTypes = {
+
+  // Sets the theme of the notification [legacy].
+  as: PropTypes.string,
+
+  // Sets the theme of the notification.
+  colourVariant: PropTypes.string,
+
+  // This component supports children.
+  children: PropTypes.string.isRequired,
+
+  // Fills the pill background with colour. When fill is false only the border is coloured.
+  fill: PropTypes.bool,
+
+  // Callback function for when the pill is clicked.
+  onClick: PropTypes.func,
+
+  // Callback function to delete the component, when the added Icon is clicked.
+  onDelete: PropTypes.func
+};
```

Comment:
I think that there should be a `disabled` prop and that we should override other colours in `pill.style.js` instead of passing a `disabled` colour as a `colourVariant`.

Reply:
`disabled` is an attribute of some form controls like input, I think we'd be better to change the name and continue to pass it as a `colourVariant` as the behaviour we're looking for is the same as for other colours.

Reply:
I checked with UX, their opinion is:

> Pills are just visual indicators with no interactivity, but sometimes they're used as links or tags added inside an input... I guess a disabled link is unusual, and tags can be deleted but not disabled. So I guess overall there may not be a use for a disabled pill, I think it's OK not to have a disabled state.

Therefore, as this is just a legacy feature that is deprecated I'd like to leave it as-is.